support-workers/src/test/scala/com/gu/zuora/ZuoraITSpec.scala

```diff
@@ -39,6 +40,24 @@ class ZuoraITSpec extends AsyncFlatSpec with Matchers {
     }
   }
 
+  it should "retrieve subscription redemption information from a redemption code" in {
+    val redemptionCode = "gd12-fnc4ox"
+    uatService.getSubscriptionFromRedemptionCode(RedemptionCode(redemptionCode).right.get).map {
+      response =>
+        response.records.size shouldBe 1
+        response.records.head.gifteeIdentityId shouldBe None
+        response.records.head.redemptionCode shouldBe Some(redemptionCode)
+    }
+  }
+
+  it should "handle invalid redemption codes" in {
+    val invalidRedemptionCode = "xxxx-0000"
```

Comment:
I'm surprised that redemption code passes the validation to get into the RedemptionCode type?  if so it shouldn't

Reply:
It does, are you thinking of the GiftCode validation? It wouldn't pass that.

Reply:
It does pass, this is a `RedemptionCode` which has different validation to a `GiftCode`. This does make me wonder if this could be an issue though - that we have two types with different validation which both go into the same custom field in Zuora.

Reply:
good point, yes we have overloaded the code field based on the value of ReaderType (or the first digit being 'g' or 'c').  This was alreaddy the intention even when matt spreadsheet was proposing the code format to be very consistent, but now we have diverged them further.
I would expect any code reading the subscription to interpret this field in conjunction with ReaderType, and for analysis in the DL they could do the same, or use the first letter.

There can't be the case where a sub is both corporate and gift, so the field would not be used at the same time.  Think of it as an `Option[Either[GiftCode,CorporateCode]]`
I can definitely see that it adds complexity to have a different data model at rest (zuora) from in transit (step functions) and we generally try to keep them as similar as possible.  This is also related to the ReaderType discussion we had as now the step function execution and zuora sub don't have a 1:1 mapping any more.